src/net_processing.cpp

```diff
@@ -2053,6 +2060,17 @@ void static ProcessOrphanTx(CConnman& connman, CTxMemPool& mempool, std::set<uin
                 // if we start doing this too early.
                 assert(recentRejects);
                 recentRejects->insert(orphanTx.GetWitnessHash());
+                // If the transaction failed for TX_INPUTS_NOT_STANDARD,
+                // then we know that the witness was irrelevant to the policy
+                // failure, since this check depends only on the txid
+                // (the scriptPubKey being spent is covered by the txid).
+                // Add the txid to the reject filter to prevent repeated
+                // processing of this transaction in the event that child
+                // transactions are later received (resulting in
+                // parent-fetching by txid via the orphan-handling logic).
+                if (orphan_state.GetResult() == TxValidationResult::TX_INPUTS_NOT_STANDARD && orphanTx.GetWitnessHash() != orphanTx.GetHash()) {
```

Comment:
why does `orphanTx.GetWitnessHash() != orphanTx.GetHash()` matter?

Reply:
Ok, because the wtxid is inserted above already, it would be a redundant entry for a non-witness-having tx?

Reply:
Yes, inserting the same hash twice degrades the filter IIRC

Reply:
Not quite. Adding the same entry is a no-op on the underlying bloom filter (since the same entry hashes to the same values). It does however increment the internal `nEntriesThisGeneration` counter, which means we'll be able to add fewer entries before wiping a previous generation. It effectively lowers the actual capacity of the rolling bloom filter.

Reply:
Added a comment to explain this.